rest/diagnostic_doc_api.go

```diff
@@ -39,3 +53,45 @@ func (h *handler) handleGetDocChannels() error {
 	h.writeJSON(resp)
 	return nil
 }
+
+// HTTP handler for a GET of a document
+func (h *handler) handleSyncFnDryRun() error {
+	docid := h.PathVar("docid")
+	body, err := h.readDocument()
+	if err != nil {
+		return err
+	}
+	_, _, channelSet, access, roles, err := h.collection.SyncFnDryrun(h.ctx(), body, docid)
```

Comment:
Since this appears to be the only caller of SyncFnDryRun, should unused omitted fields just be removed from the method?  (I think we actually want to use expiry, but SyncFnDryrun doesn't need to return oldBody.